src/AuthorizationService.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization;
+
+use Authorization\Policy\BeforePolicyInterface;
+use Authorization\Policy\Exception\MissingMethodException;
+use Authorization\Policy\ResolverInterface;
+
+class AuthorizationService implements AuthorizationServiceInterface
+{
+    /**
+     * Authorization policy resolver.
+     *
+     * @var \Authorization\Policy\ResolverInterface
+     */
+    protected $resolver;
+
+    /**
+     *
+     * @param \Authorization\Policy\ResolverInterface $resolver Authorization policy resolver.
+     */
+    public function __construct(ResolverInterface $resolver)
+    {
+        $this->resolver = $resolver;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function can(IdentityInterface $user, $action, $resource)
+    {
+        $policy = $this->resolver->getPolicy($resource);
+
+        if ($policy instanceof BeforePolicyInterface) {
+            if (!$policy->before($user, $resource)) {
```

Comment:
Should we only fail early if the `before` method returns false? That would allow

* `true` to skip the can method.
* `null` to fall through.
* `false` to abort early

 The use case I can think of for `true` is an administrator account which is allowed to do everything in the system.